File: Wallet.java
import java.util.Scanner;

public class Wallet {
    private double balance;

    public Wallet() {
        this.balance = 0;
    }

    public void deposit(double amount) {
        this.balance += amount;
        System.out.println("Dépôt réussi");
    }

    public void withdraw(double amount) {
        if (amount <= this.balance) {
            this.balance -= amount;
            System.out.println("Retrait réussi");
        } else {
            System.out.println("Fonds insuffisants");
        }
    }

    public double getBalance() {
        return this.balance;
    }

    public void transfer(double amount, Wallet targetWallet) {
        if (amount <= this.balance) {
            this.balance -= amount;
            targetWallet.deposit(amount);
            System.out.println("Transfert réussi");
        } else {
            System.out.println("Fonds insuffisants pour effectuer le transfert");
        }
    }

    public static void main(String[] args) {
        Wallet myWallet1 = new Wallet();
        Wallet myWallet2 = new Wallet();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("Menu :");
            System.out.println("1. Déposer de l'argent");
            System.out.println("2. Retirer de l'argent");
            System.out.println("3. Consulter le solde");
            System.out.println("4. Transférer de l'argent");
            System.out.println("0. Quitter");

            System.out.print("Choisissez une option : ");
            int choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Entrez le montant à déposer : ");
                    double depositAmount = scanner.nextDouble();
                    myWallet1.deposit(depositAmount);
                    break;
                case 2:
                    System.out.print("Entrez le montant à retirer : ");
                    double withdrawAmount = scanner.nextDouble();
                    myWallet1.withdraw(withdrawAmount);
                    break;
                case 3:
                    System.out.println("Solde actuel : " + myWallet1.getBalance());
                    break;
                case 4:
                    System.out.print("Entrez le montant à transférer : ");
                    double transferAmount = scanner.nextDouble();
                    myWallet1.transfer(transferAmount, myWallet2);
                    break;
                case 0:
                    System.out.println("Au revoir !");
                    System.exit(0);
                default:
                    System.out.println("Option invalide. Veuillez choisir une option valide.");
            }
        }
    }
}

File: wallet.php
<?php

class Wallet {
    private $balance;

    public function __construct() {
        $this->balance = 0;
    }

    public function deposit($amount) {
        $this->balance += $amount;
        echo "Dépôt réussi\n";
    }

    public function withdraw($amount) {
        if ($amount <= $this->balance) {
            $this->balance -= $amount;
            echo "Retrait réussi\n";
        } else {
            echo "Fonds insuffisants\n";
        }
    }

    public function getBalance() {
        return $this->balance;
    }

    public function transfer($amount, Wallet $targetWallet) {
        if ($amount <= $this->balance) {
            $this->balance -= $amount;
            $targetWallet->deposit($amount);
            echo "Transfert réussi\n";
        } else {
            echo "Fonds insuffisants pour effectuer le transfert\n";
        }
    }
}

$myWallet1 = new Wallet();
$myWallet2 = new Wallet();

while (true) {
    echo "Menu :\n";
    echo "1. Déposer de l'argent\n";
    echo "2. Retirer de l'argent\n";
    echo "3. Consulter le solde\n";
    echo "4. Transférer de l'argent\n";
    echo "0. Quitter\n";

    $choice = readline("Choisissez une option : ");

    switch ($choice) {
        case '1':
            $depositAmount = readline("Entrez le montant à déposer : ");
            $myWallet1->deposit((float)$depositAmount);
            break;
        case '2':
            $withdrawAmount = readline("Entrez le montant à retirer : ");
            $myWallet1->withdraw((float)$withdrawAmount);
            break;
        case '3':
            echo "Solde actuel : " . $myWallet1->getBalance() . "\n";
            break;
        case '4':
            $transferAmount = readline("Entrez le montant à transférer : ");
            $myWallet1->transfer((float)$transferAmount, $myWallet2);
            break;
        case '0':
            echo "Au revoir !\n";
            exit(0);
        default:
            echo "Option invalide. Veuillez choisir une option valide.\n";
    }
}
